managing-os/windows/windows-vm-test.md

```diff
@@ -0,0 +1,306 @@
+qvm-prefs -s win7new debug false
+~~~
+
+### Detailed instructions ###
+
+MS Windows versions considerations:
+
+- The instructions *may* work on other versions than Windows 7 x64 but haven't been tested.
+- Qubes Windows Tools (QWT) only supports Windows 7 x64.
+
+Create a VM named win7new in [HVM](/doc/hvm/) mode (Xen's current PVH limitations precludes from using PVH):
+
+~~~
+qvm-create win7new --hvm --label red
+~~~
+
+Windows' installer requires a significant amount of memory or else the VM will crash with such errors:
+
+`/var/log/xen/console/hypervisor.log`:
+
+> p2m_pod_demand_populate: Dom120 out of PoD memory! (tot=102411 ents=921600 dom120)
+> (XEN) domain_crash called from p2m-pod.c:1218
+> (XEN) Domain 120 (vcpu#0) crashed on cpu#3:
+
+So, increase the VM's memory to 4096MB (memory = maxmem because we don't use memory balancing).
+
+~~~
+qvm-prefs -s win7new memory 4096
+qvm-prefs -s win7new maxmem 4096
+~~~
+
+A typical Windows 7 installation requires between 15GB up to 19GB of disk space depending on the version (Home/Professional/...). Windows updates also end up using significant space. So, extend the root volume from the default 10GB to 25GB (note: it is straightforward to increase the root volume size after Windows is installed: simply extend the volume again in dom0 and then extend the system partition with Windows's disk manager).
+
+~~~
+qvm-grow-root win7new 25g 
+~~~
+
+Set the debug flag in order to have a graphical console:
+
+~~~
+qvm-prefs -s win7new debug true
+~~~
+
+The second part of the installation process will crash with the standard VGA video adapter and the VM will stay in "transient" mode with the following error in `guest-win7new-dm.log`:
+
+> qemu: /home/user/qubes-src/vmm-xen-stubdom-linux/build/qemu/exec.c:1187: cpu_physical_memory_snapshot_get_dirty: Assertion `start + length <= snap->end' failed.
+
+To avoid that error we temporarily have to switch the video adapter to 'cirrus'. Backup the VM's configuration file and substitute the video driver from 'xen' to 'cirrus':
+
+~~~
+cp /var/lib/qubes/appvms/win7new/win7new.conf /tmp
+sed -i "s/<model \+type='xen' \+vram=/<model type='cirrus' vram=/" /var/lib/qubes/appvms/win7new/win7new.conf
+# or edit the file manually ; either way, make sure the adapter is now cirrus.
+~~~
+
+The VM is now ready to be started; the best practice is to use an installation ISO [located in a VM](/doc/hvm/#installing-an-os-in-an-hvm-domain):
+
+~~~
+qvm-start --cdrom=untrusted:/home/user/windows_install.iso win7new
+~~~
+
+Given the higher than usual memory requirements of Windows, you may get a `Not enough memory to start domain 'win7new'` error. In that case try to free memory by shutdowning unneeded VMs before starting the Windows VM.
+
+At this point you may open a few tabs in dom0 for debugging should something go amiss: 
+
+~~~
+tail -f /var/log/qubes/vm-win7new.log
+tail -f /var/log/xen/console/hypervisor.log
+tail -f /var/log/xen/console/guest-win7new-dm.log
```

Comment:
Hint: you can pass multiple files to `tail -f`, then you'll have nice monitoring of all those files in one window.